Allow requesting html or text responses for directory listings over http

creationix opened the pull request:
This PR implements #71 

The only behavioral changes are:

If no `Accept` header is sent by the client, it will render as JSON text (before it always rendered html)
If you do specify a format and it's on of the 3 supported formats, you'll get that format.
I added a newline to the end of the html format to make it CLI friendly.

The `text/plain` format is simply the filenames with a trailing slash on folders to signify it's a folder.

For example:

```http
imac:hashbase tim$ curl -i http://localhost:8080/ -H 'Host: fritter-admin.imac.local' -H 'Accept: text/plain'
HTTP/1.1 200 OK
X-Powered-By: Express
X-RateLimit-Limit: 100
X-RateLimit-Remaining: 99
Content-Type: text/plain
Date: Mon, 08 Jan 2018 20:58:34 GMT
Connection: keep-alive
Transfer-Encoding: chunked

posts/
votes/
avatar.png
dat.json
profile.json
```

The `applicaiton/json` format is pretty close to the stat entries returned by hyperdrive, but only includes the interesting bits as well as the full path with trailing slash for directories:

```http
imac:hashbase tim$ curl -i http://localhost:8080/ -H 'Host: fritter-admin.imac.local' -H 'Accept: application/json'
HTTP/1.1 200 OK
X-Powered-By: Express
X-RateLimit-Limit: 100
X-RateLimit-Remaining: 99
Content-Type: application/json
Date: Mon, 08 Jan 2018 20:59:34 GMT
Connection: keep-alive
Transfer-Encoding: chunked

[
  {
    "path": "/posts/",
    "name": "posts",
    "type": "directory",
    "mtime": 1515184134107
  },
  {
    "path": "/votes/",
    "name": "votes",
    "type": "directory",
    "mtime": 1515184134108
  },
  {
    "path": "/avatar.png",
    "name": "avatar.png",
    "type": "file",
    "mtime": 1515184134000,
    "size": 150859,
    "blocks": 3,
    "downloaded": 3
  },
  {
    "path": "/dat.json",
    "name": "dat.json",
    "type": "file",
    "mtime": 1515184133681,
    "size": 187,
    "blocks": 1,
    "downloaded": 1
  },
  {
    "path": "/profile.json",
    "name": "profile.json",
    "type": "file",
    "mtime": 1515433706000,
    "size": 1700,
    "blocks": 1,
    "downloaded": 1
  }
]
```

pfrazee commented on lib/apis/archive-files.js:
```diff
@@ -141,10 +142,36 @@ module.exports = class ArchiveFilesAPI {
     if (entry && entry.isDirectory()) {
-      res.writeHead(200, 'OK', {
-        'Content-Type': 'text/html'
+      var negotiator = new Negotiator(req)
```
What about https://expressjs.com/en/4x/api.html#req.accepts or https://expressjs.com/en/4x/api.html#res.format ? Do they mishandle accept headers?

creationix replied:
I wasn't aware of those, they would probably work fine.

pfrazee commented on lib/apis/archive-files.js:
```diff
@@ -141,10 +142,36 @@ module.exports = class ArchiveFilesAPI {
+      // If the client asked for text or didn't specify serve text
+      if (type === 'text/plain') {
+        res.writeHead(200, 'OK', {
+          'Content-Type': 'text/html'
```
Would this need to be text/plain?

creationix replied:
Yes, typo.

creationix commented:
I updated the PR.  I no longer adds a new dependency, I fixed the typos and it defaults to JSON instead of plain text in case no `Accept` header is in the request.

pfrazee commented:
👍  deploying now

creationix commented:
It works!

```http
imac:~ tim$ curl -i https://fritter-creationix.hashbase.io/
HTTP/1.1 200 OK
Server: nginx/1.6.2
Date: Mon, 08 Jan 2018 21:59:58 GMT
Content-Type: application/json
Connection: close
Vary: Accept-Encoding
X-Powered-By: Express
X-RateLimit-Limit: 100
X-RateLimit-Remaining: 98

[
  {
    "path": "/posts/",
    "name": "posts",
    "type": "directory",
    "mtime": 1515184134107
  },
  {
    "path": "/votes/",
    "name": "votes",
    "type": "directory",
    "mtime": 1515184134108
  },
  {
    "path": "/avatar.png",
    "name": "avatar.png",
    "type": "file",
    "mtime": 1515184134000,
    "size": 150859,
    "blocks": 3,
    "downloaded": 3
  },
  {
    "path": "/dat.json",
    "name": "dat.json",
    "type": "file",
    "mtime": 1515184133681,
    "size": 187,
    "blocks": 1,
    "downloaded": 1
  },
  {
    "path": "/profile.json",
    "name": "profile.json",
    "type": "file",
    "mtime": 1515433706000,
    "size": 1700,
    "blocks": 1,
    "downloaded": 1
  }
]
```

creationix commented:
```http
imac:~ tim$ curl -i https://fritter-creationix.hashbase.io/ -H 'Accept: text/plain'
HTTP/1.1 200 OK
Server: nginx/1.6.2
Date: Mon, 08 Jan 2018 22:00:45 GMT
Content-Type: text/plain
Connection: close
Vary: Accept-Encoding
X-Powered-By: Express
X-RateLimit-Limit: 100
X-RateLimit-Remaining: 99

posts/
votes/
avatar.png
dat.json
profile.json
```
